Library/Homebrew/cmd/upgrade.rb

```diff
@@ -44,7 +44,7 @@ def upgrade
 
     outdated.each do |f|
       upgrade_formula(f)
-      cleanup_formula(f) if ARGV.include?("--cleanup") && f.installed?
+      Homebrew::Cleanup.cleanup_formula f if ARGV.include?("--cleanup") && f.installed?
```

Comment:
In this case I'd do `next unless ARGV.include?("--cleanup")` and `next unless f.installed?` to split this onto a few more lines.